frontend/src/pages/index.js

```diff
@@ -1,49 +1,89 @@
-import React, { Component } from 'react'
+import React from 'react'
+import { Component } from 'react'
 import Router from 'next/router'
-import { Button, Container, Input, Row, Col } from 'reactstrap'
-import { validateKey } from '../utils/api'
-import ReactLoading from 'react-loading'
+import { loginUser, loginGoogleUser } from '../utils/api'
 import Nav from '../components/nav'
 import Head from '../components/head'
+import { GoogleLogin } from 'react-google-login'
+import {
+  Container,
+  Form,
+  Button,
+  FormGroup,
+  Label,
+  Input,
+  Card,
+  CardBody,
+  CardTitle,
+  Modal,
+  ModalBody,
+  ModalHeader,
+  ModalFooter
+} from 'reactstrap'
+import cookie from 'js-cookie'
 
 class LoginPage extends Component {
   constructor(props) {
     super(props)
     this.state = {
-      currentKey: '',
-      loading: false
+      email: '',
+      password: '',
+      errorMessage: '',
+      showInvalidRequestModal: false
     }
   }
 
-  handleSubmit = async () => {
-    this.setState({
-      loading: true
-    })
-    const { success, result } = await validateKey(this.state.currentKey)
-    if (success) {
-      localStorage.setItem('interviewerKey', this.state.currentKey)
-      localStorage.setItem('interviewerName', result.name)
-      this.setState({
-        loading: false
+  // use cookie to hold information that is valid across the whole site
+  setCookie = (key, value) => {
+    const cookieExpirationDays = 1
+    if (process.browser) {
+      cookie.set(key, value, {
+        expires: cookieExpirationDays,
+        path: '/'
       })
-      Router.push('/dashboard')
+    }
+  }
+
+  handleGoogle = async e => {
+    const memberKey = 'ohno'
+    const result = await loginGoogleUser(e.tokenId)
+    const resp = await result.json()
+    if (!resp.success) {
+      this.setState({ errorMessage: resp.message, showInvalidRequestModal: true })
+      console.log(resp.message)
     } else {
-      this.setState({
-        loading: false
-      })
-      alert("Couldn't log in. Is your key correct?")
+      // set token value so google can access it
+      this.setCookie('token', e.tokenId)
+      // set google to true so server knows to send the request to google
+      this.setCookie('google', true)
+      // set localStorage value so it's valid across the whole site
+      localStorage.setItem('interviewerKey', memberKey) // TODO: Create switch statements for roles - Issue #314
+      Router.push('/dashboard')
     }
   }
 
-  onTextChange = e => {
-    this.setState({ currentKey: e.target.value })
+  handleChange = event => {
+    const value = event.target.value
+    const name = event.target.name
+    this.setState({ [name]: value })
   }
 
-  // handles when user presses "Enter" when input is focused
-  _handleKeyPress = e => {
-    if (e.key === 'Enter') {
-      this.handleSubmit()
-    }
+  handleSubmit = () => {
+    const memberKey = 'ohno'
```

Comment:
so if you move this to the top as `MEMBER_KEY` you won't have to redefine it in both `handleGoogle` and `handleSubmit`

Reply:
(you can handle this in a follow-up PR so we can ship this one faster)